packages/app/src/SessionEventList.tsx

```diff
@@ -0,0 +1,281 @@
+  eventSource: 'navigation' | 'chat' | 'network' | 'custom';
+  title: string;
+  description: string;
+  timestamp: Date;
+  formattedTimestamp: string;
+  duration: number;
+};
+
+const EVENT_ROW_SOURCE_ICONS = {
+  navigation: 'bi bi-geo-alt',
+  chat: 'bi bi-chat-dots',
+  network: 'bi bi-arrow-left-right',
+  custom: 'bi bi-cursor',
+};
+
+const EventRow = React.forwardRef(
+  (
+    {
+      dataIndex,
+      event,
+      isHighlighted,
+      onClick,
+      onTimeClick,
+    }: {
+      dataIndex: number;
+      event: SessionEvent;
+      isHighlighted: boolean;
+      onClick: VoidFunction;
+      onTimeClick: VoidFunction;
+    },
+    ref: React.Ref<HTMLDivElement>,
+  ) => {
+    return (
+      <div
+        data-index={dataIndex}
+        ref={ref}
+        className={cx(styles.eventRow, {
+          [styles.eventRowError]: event.isError,
+          [styles.eventRowSuccess]: event.isSuccess,
+          [styles.eventRowHighlighted]: isHighlighted,
+        })}
+      >
+        <div className={styles.eventRowIcon}>
+          <i
+            className={
+              EVENT_ROW_SOURCE_ICONS[event.eventSource] || 'bi bi-terminal'
+            }
+          />
+        </div>
+        <div className={styles.eventRowContent} onClick={onClick}>
+          <div className={styles.eventRowTitle}>
+            {event.title}{' '}
+            {event.duration > 0 && <span>{event.duration}ms</span>}
+          </div>
+          <div className={styles.eventRowDescription} title={event.description}>
+            {event.description}
+          </div>
+        </div>
+        <div className={styles.eventRowTimestamp} onClick={onTimeClick}>
+          <i className="bi bi-play-fill me-1 fs-8" />
+          {event.formattedTimestamp}
+        </div>
+      </div>
+    );
+  },
+);
+
+export const SessionEventList = ({
+  queriedConfig,
+  onClick,
+  onTimeClick,
+  focus,
+  minTs,
+  showRelativeTime,
+  eventsFollowPlayerPosition,
+}: {
+  queriedConfig: ChartConfigWithOptDateRange;
+  // highlightedResultId: string | undefined;
+  focus: { ts: number; setBy: string } | undefined;
+  minTs: number;
+  showRelativeTime: boolean;
+  onClick: (logId: string, sortKey: string) => void;
+  onTimeClick: (ts: number) => void;
+  eventsFollowPlayerPosition: boolean;
+}) => {
+  const { data, isLoading, isError, error, isPlaceholderData, isSuccess } =
+    useQueriedChartConfig(queriedConfig, {
+      placeholderData: (prev: any) => prev,
+      queryKey: ['SessionEventList', queriedConfig],
+    });
+
+  const formatTime = useFormatTime();
+
+  const events = data?.data ?? [];
+
+  const rows = React.useMemo(() => {
+    return (
+      events.map((event, i) => {
+        const { timestamp, durationInMs } = event;
+
+        // TODO: we should just use timestamp and durationInMs instead of startOffset and endOffset
+        const startOffset = new Date(timestamp).getTime();
+        const endOffset = new Date(startOffset).getTime() + durationInMs;
```

Comment:
only change here that inits startOffset and endOffset using new timestamp and durationInMs